frontend/integration-tests/tests/crud.scenario.ts

```diff
@@ -10,9 +10,13 @@ import { appHost, testName, checkLogs, checkErrors } from '../protractor.conf';
 import * as crudView from '../views/crud.view';
 import * as yamlView from '../views/yaml.view';
 import * as namespaceView from '../views/namespace.view';
+import { K8sKind } from '../../public/module/k8s';
+import { VirtualMachineModel } from '../../public/models';
 
 const K8S_CREATION_TIMEOUT = 15000;
 
+const getKindForCRD = (model: K8sKind) => `${model.apiGroup}~${model.apiVersion}~${model.kind}`;
```

Comment:
Importing the `referenceForModel` function here:

```ts
import { referenceForModel } from '../../public/module/k8s';
```

and calling it causes the following error:

```
E/launcher - Error: /home/vszocs/go/src/github.com/openshift/console/frontend/node_modules/lodash-es/lodash.js:10
export { default as add } from './add.js';
^^^^^^

SyntaxError: Unexpected token export
    at new Script (vm.js:79:7)
    at createScript (vm.js:251:10)
    at Object.runInThisContext (vm.js:303:10)
    at Module._compile (internal/modules/cjs/loader.js:656:28)
    at Module._extensions..js (internal/modules/cjs/loader.js:699:10)
    at Object.require.extensions.(anonymous function) [as .js] (/home/vszocs/go/src/github.com/openshift/console/frontend/node_modules/ts-node/src/index.ts:392:14)
    at Module.load (internal/modules/cjs/loader.js:598:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:537:12)
    at Function.Module._load (internal/modules/cjs/loader.js:529:3)
    at Module.require (internal/modules/cjs/loader.js:636:17)
```

In `public/module/k8s/k8s` we do `import * as _ from 'lodash-es'` and in `lodash-es/lodash.js` there's ES6 style export:

```js
export { default as add } from './add.js'
```

which causes this error. This is the reason why didn't use the `referenceForModel` function.

I can look into this problem and try to fix it.

Reply:
OK, I see now that integration tests actually use `lodash` instead of `lodash-es` :smiley:

I've tried to use `-I 'node_modules/(?!lodash-es/.*)'` (custom ignore pattern) argument for `ts-node` within the `test-suite` script, but ran into some weird issues.

So for integration tests specifically, we'll need to map `lodash-es` to `lodash` in terms of TypeScript module resolution.

Reply:
I've solved the issue of aliasing `lodash-es` to `lodash` in context of integration tests.

Doing that, I've run into another issue - access to `window` global which is `undefined` when being evaluated in context of integration tests (Jasmine) as part of the import process, for example:

```ts
// public/module/k8s/k8s.ts
export const k8sBasePath = `${(window as any).SERVER_FLAGS.basePath}api/kubernetes`;
```

I've solved that issue as well, mocking `window.SERVER_FLAGS` object as necessary.

It's now possible to `import` any code from `public` directory within integration test code, so I'll go ahead and use the `referenceForModel` function as suggested by @spadgett.